lib/ts/recipe/passwordless/recipeImplementation.ts

```diff
@@ -0,0 +1,227 @@
+                    preAuthSessionId: input.preAuthSessionId,
+                };
+            }
+
+            return await querier.post("/signinup/code/consume", { body: JSON.stringify(bodyObj) }, (context) => {
+                return input.config.preAPIHook({
+                    ...context,
+                    action: "PASSWORDLESS_CONSUME_CODE",
+                });
+            });
+        },
+        doesEmailExist: async function (input: { email: string; config: NormalisedConfig }): Promise<boolean> {
+            const response: ExistsAPIResponse = await querier.get(
+                "/signup/email/exists",
+                {},
+                { email: input.email },
+                (context) => {
+                    return input.config.preAPIHook({
+                        ...context,
+                        action: "EMAIL_EXISTS",
+                    });
+                }
+            );
+
+            return response.exists;
+        },
+
+        doesPhoneNumberExist: async function (input: {
+            phoneNumber: string;
+            config: NormalisedConfig;
+        }): Promise<boolean> {
+            const response: ExistsAPIResponse = await querier.get(
+                "/signup/phoneNumber/exists",
+                {},
+                { phoneNumber: input.phoneNumber },
+                (context) => {
+                    return input.config.preAPIHook({
+                        ...context,
+                        action: "PHONE_NUMBER_EXISTS",
+                    });
+                }
+            );
+
+            return response.exists;
+        },
+        getLoginAttemptInfo: function ():
+            | undefined
+            | {
+                  deviceId: string;
+                  preAuthSessionId: string;
+                  flowType: FlowType;
+                  contactInfo: string;
+                  contactInfoType: "EMAIL" | "PHONE";
+                  lastResend: number;
+              } {
+            const storedInfo = localStorage.getItem(PASSWORDLESS_LOGIN_ATTEMPT_INFO_STORAGE_KEY);
+            if (!storedInfo) {
+                return undefined;
+            }
+            try {
+                const info = JSON.parse(storedInfo);
+
+                return {
+                    contactInfo: info.contactInfo,
+                    contactInfoType: info.contactInfoType,
+                    deviceId: info.deviceId,
+                    flowType: info.flowType,
+                    preAuthSessionId: info.preAuthSessionId,
+                    lastResend: info.lastResend,
+                };
+            } catch (ex) {
+                return undefined;
+            }
+        },
+        setLoginAttemptInfo: function (input: {
+            deviceId: string;
+            preAuthSessionId: string;
+            flowType: FlowType;
+            contactInfo: string;
+            contactInfoType: "EMAIL" | "PHONE";
+        }): void {
+            localStorage.setItem(
```

Comment:
Do not use `localstorage` directly. Instead, there is a function in ts/utils called `setLocalStorage`